x-pack/plugins/data_visualizer/public/application/common/components/stats_table/utils.ts

```diff
@@ -36,3 +37,55 @@ export const getTFPercentage = (config: FileBasedFieldVisConfig) => {
     falseCount,
   };
 };
+
+export const calculateTableColumnsDimensions = (width?: number) => {
+  const defaultSettings = {
+    expander: '40px',
+    type: '75px',
+    docCount: '225px',
+    distinctValues: '225px',
+    distributions: '225px',
+    showIcon: true,
+    breakPoint: 'xl',
+  };
+  if (width === undefined) return defaultSettings;
+  const breakPoint = getBreakpoint(width);
+  switch (breakPoint) {
+    case 'xs':
+    case 's':
+      return {
+        expander: '25px',
+        type: '40px',
+        docCount: 'auto',
+        distinctValues: 'auto',
+        distributions: 'auto',
+        showIcon: false,
+        breakPoint,
+      };
+
+    case 'm':
+    case 'l':
+      return {
+        expander: '25px',
+        type: '40px',
+        docCount: 'auto',
+        distinctValues: 'auto',
+        distributions: 'auto',
+        showIcon: false,
+        breakPoint,
+      };
+
+    default:
+      return defaultSettings;
+  }
+};
+
+export const getKibanaFieldType = (fieldType: string) => {
```

Comment:
Can you use `JobFieldType` instead of `string` here?

Reply:
I removed this for now as this is redundant for this PR https://github.com/elastic/kibana/pull/112870/commits/6c0c8505076c4152d0ddacfb34b3909598c08403